pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -350,6 +350,7 @@ func (c *Controller) etcdSaveWithBackoff(b *v3.EtcdBackup) (runtime.Object, erro
 		if err != nil {
 			return b, err
 		}
+		cluster.Spec.RancherKubernetesEngineConfig.Services.Etcd.BackupConfig = &b.Spec.BackupConfig
```

Comment:
I'm a bit unfamiliar with this code - the following questions encompass a bit more than the code that you changed:
- Is the idea with this change that when the backupDriver does the ETCDSave, it (in this case) was missing the backup config, causing an error? Therefore, in order to fix that, you temporarily (I say temporarily because I don't think that you update the cluster) set the backup config of the cluster to this backupConfig (both on the create/remove functions)?
- Is this functionality exclusive to RKE (I ask b/c this is a property of the RancherKubernetesEngineConfig)?
- The Create function appears to check that something is set in this value before calling these functions using the `isBackupSet` function. Is this change potentially overwriting another value?

Reply:
> Is the idea with this change that when the backupDriver does the ETCDSave, it (in this case) was missing the backup config

It was not missing the backup config. The credential from the backup config is never stored, and so it needs to be reassembled on the object before using it. We reassemble the object [here](https://github.com/rancher/rancher/blob/58e374f9eaa0e96a76298259734348a377b1a186/pkg/controllers/management/etcdbackup/etcdbackup.go#L251) and the backup config object that we eventually use here comes from that [reassembled cluster object](https://github.com/rancher/rancher/blob/58e374f9eaa0e96a76298259734348a377b1a186/pkg/controllers/management/etcdbackup/etcdbackup.go#L255).

> I say temporarily because I don't think that you update the cluster

Yes, we never store the credential on the cluster object any more

> Is this functionality exclusive to RKE

Yes

> The Create function appears to check that something is set in this value before calling these functions using the isBackupSet function. Is this change potentially overwriting another value?

No. For one, `isBackupSet` causes `Create` to [return](https://github.com/rancher/rancher/blob/58e374f9eaa0e96a76298259734348a377b1a186/pkg/controllers/management/etcdbackup/etcdbackup.go#L99) before [this code is called](https://github.com/rancher/rancher/blob/58e374f9eaa0e96a76298259734348a377b1a186/pkg/controllers/management/etcdbackup/etcdbackup.go#L120) if there is nothing set. For another, the value it is setting comes from the [reassembled cluster itself](https://github.com/rancher/rancher/blob/58e374f9eaa0e96a76298259734348a377b1a186/pkg/controllers/management/etcdbackup/etcdbackup.go#L255). What this is doing is overcoming the fact that the original cluster object was dropped and [re-fetched](https://github.com/rancher/rancher/blob/58e374f9eaa0e96a76298259734348a377b1a186/pkg/controllers/management/etcdbackup/etcdbackup.go#L349) in this function.

Reply:
Ok - I see. So we re-fetch the cluster, which removes the backupconfig secrets options. Since we set this value on the backup earlier in another function, we just re-fetch it from the object (where we have it saved). 

Do we need to do something similar for the remove call? I don't see an assemble S3 credential call anywhere in the call chain for removing the backup.